src/graphics/program-lib/chunks/TBNderivative.frag

```diff
@@ -15,6 +15,7 @@ void getTBN() {
     vec3 B = dp2perp * duv1.y + dp1perp * duv2.y;
 
     // construct a scale-invariant frame
-    float invmax = 1.0 / sqrt( max( dot(T,T), dot(B,B) ) );
+    float denom = max( dot(T,T), dot(B,B) );
+    float invmax = (denom == 0.0) ? 0.0 : 1.0 / sqrt( denom );
```

Comment:
I wonder if there's a workaround without branching?

Reply:
I don't think the branch is an issue. The graphics card will calculate both sides and choose based on comparison (not actually perform a branch and jump). So the overhead introduced here is the comparison. If you can think of a better way though, I'm all ears.

Reply:
yep I agree with this, should be fine.